script/tool/test/format_command_test.dart

```diff
@@ -348,4 +494,95 @@ void main() {
           contains('Unable to determine diff.'),
         ]));
   });
+
+  test('Batches moderately long file lists on Windows', () async {
+    mockPlatform.isWindows = true;
+
+    const String pluginName = 'a_plugin';
+    // -1 since the command itself takes some length.
+    const int batchSize = (windowsCommandLineMax ~/ 100) - 1;
+
+    // Make the file list one file longer than would fit in the batch.
+    final List<String> batch1 =
+        _get99CharacterPathExtraFiles(pluginName, batchSize + 1);
```

Comment:
No need to `-1` batchSize above, then?

Reply:
I left it this way deliberately, because I think it's easier to understand the intention: figure out the batch size (which requires -1) then overflow it by one (which requires +1). Otherwise it feels like it just sort of works by accident.

It's also easier to see how this would scale. E.g., if we end up needing a test with N batches in the future, having the correct batch size in the code that would be copied from makes that easier.